File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        // 정답 선언 및 초기화
        int answer = 0;
        // n의 홀짝 여부에 따라 다른 연산을 할 for문
        for(int i=1; i<=n; i++){
            if(n%2 == 0 && i%2 == 0){
                answer += i*i;
            } else if(n%2 == 1 && i%2 ==1){
                answer += i;
            }
        }
        
        // 정답 리턴
        return answer;
    }
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    answer = 0
    i = 1
    even = 0
    odd = 0
    
    while i <= n:
        if i % 2 != 0:
            odd = odd + i
        else:
            even = even + (i * i)
        i = i + 1
        
    if n % 2 != 0:
        answer = odd
    else:
        answer = even
                
    return answer